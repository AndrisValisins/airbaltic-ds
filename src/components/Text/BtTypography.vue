<template>
  <component :is="variant">
    <slot />
  </component>
</template>

<script lang="ts">
import { defineComponent } from "vue";

export default defineComponent({
  name: "BaseTypography",
  props: {
    variant: {
      type: String,
      default: "p",
      validator(value: string) {
        return ["h1", "h2", "h3", "h4", "h5", "h6", "p"].includes(value);
      },
    },
  },
});
</script>

<style lang="scss">
h1 {
  color: red;
}
</style>
