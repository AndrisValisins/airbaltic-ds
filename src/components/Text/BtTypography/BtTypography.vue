<template>
  <component :is="variant">
    <slot />
  </component>
</template>

<script lang="ts" setup>
defineProps({
  variant: {
    type: String,
    default: "p",
    validator(value: string) {
      return ["h1", "h2", "h3", "h4", "h5", "h6", "p"].includes(value);
    },
  },
});
</script>

<style lang="scss">
h1 {
  color: red;
}
</style>
