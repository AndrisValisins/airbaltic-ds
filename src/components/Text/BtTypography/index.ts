export { default as BtTypography } from "./BtTypography.vue";
