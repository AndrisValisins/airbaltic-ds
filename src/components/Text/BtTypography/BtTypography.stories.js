import BtTypography from "./BtTypography.vue";

export default {
  title: "Atoms/Text",
  component: BtTypography,
  argTypes: {
    variant: {
      control: { type: "select" },
      options: ["h1", "h2", "h3", "h4", "h6", "body"],
    },
  },
};

const Template = (args) => ({
  components: { BtTypography },
  setup() {
    return { args };
  },
  template: `<bt-typography v-bind="args">
    <template #default>{{ args.default }}</template>
  </bt-typography>`,
});

export const Body = Template.bind({});
Body.args = {
  default: "Body text",
  variant: "body",
};

export const H1 = Template.bind({});
H1.args = {
  default: "Heading 1",
  variant: "h1",
};

export const H2 = Template.bind({});
H2.args = {
  default: "Heading 2",
  variant: "h2",
};

export const H3 = Template.bind({});
H3.args = {
  default: "Heading 3",
  variant: "h3",
};

export const H4 = Template.bind({});
H4.args = {
  default: "Heading 4",
  variant: "h4",
};

export const H5 = Template.bind({});
H5.args = {
  default: "Heading 5",
  variant: "h5",
};

export const H6 = Template.bind({});
H6.args = {
  default: "Heading 6",
  variant: "h6",
};
