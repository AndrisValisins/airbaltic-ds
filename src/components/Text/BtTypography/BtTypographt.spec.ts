import { describe, it, expect } from "vitest";

import { mount } from "@vue/test-utils";
import BtTypography from "./BtTypography.vue";

describe("BtTypography", () => {
  it("Matches to snapshot", () => {
    const wrapper = mount(BtTypography, {
      slots: {
        default: "Body text",
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("renders  h1 properly", () => {
    const wrapper = mount(BtTypography, {
      slots: {
        default: "Heading 1",
      },
      props: {
        variant: "h1",
      },
    });
    expect(wrapper.text()).toContain("Heading 1");
  });

  it("renders body properly", () => {
    const wrapper = mount(BtTypography, {
      slots: {
        default: "Body",
      },
    });
    expect(wrapper.text()).toContain("Body");
  });
});
