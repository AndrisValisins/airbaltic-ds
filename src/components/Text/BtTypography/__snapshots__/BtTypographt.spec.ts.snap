// Vitest Snapshot v1

exports[`BtTypography > Matches to snapshot 1`] = `
<p>
  
  Body text
  
</p>
`;
